### ISSUE TITLE
Override rendering of built-in blocks (Link)

### ISSUE THREAD

**Issue by adamtowerz** [TypeCellOS/BlockNote#1118]
**Is your feature request related to a problem? Please describe.**
I'd like to override React rendering for links, but without completely re-implementing the Link code as a custom block.

**Describe the solution you'd like**
Ideally some way to provide a `render` that returns custom JSX

**Describe alternatives you've considered**
Considered re-implementing the Link code and disabling the existing default block but I want most of the same behavior, just custom rendering.

**Comment by YousefED** [comment]
can you explain about why you want this exactly? what would you like to
build from an end user perspective?

**Comment by adamtowerz** [comment]
I'd like to check to see if a link is an internal link to a different page in the application rather than an external link, and if it is support additional features on the link like a popover on hover to peek into the page details. However, I'd like these to really just be _links_ with respect to how they're stored and handled for all other purposes.

**Comment by ClementEXWiki** [comment]
I'd love to have the same capability, my goal currently is to provide a tooltip for "internal" links as well as a logo suffixing the link's label to clearly indicate it's an internal reference.

Apart from making a custom inline block, I'm not sure how it can be done.

**Comment by ChromeGG** [comment]
Yeah, same issue here. I would like to use my component library `Link`. I tried to override the default implementation, but failed 😬

**Comment by matthewlipski** [comment]
Unfortunately, due to the way that links work in BlockNote, allowing for rendering customization using the custom inline content API would be a pretty major breaking change. If it's useful in your case though, #2666 adds the option to add HTML attributes & custom click handling to links.
